File: tests/fixtures/scala/sample.scala
import scala.math.{PI, sqrt}
import scala.collection.mutable.ArrayBuffer

def greet(name: String): String = s"Hello, $name!"

def add(a: Int, b: Int): Int = a + b

trait Shape {
  def area(): Double
  def perimeter(): Double
}

class Circle(val radius: Double) extends Shape {
  def area(): Double = PI * radius * radius
  def perimeter(): Double = 2 * PI * radius
}

object MathUtils {
  def square(x: Int): Int = x * x

  def clamp(value: Int, min: Int, max: Int): Int =
    if value < min then min
    else if value > max then max
    else value
}

val version = "1.0.0"
var counter = 0

File: tests/fixtures/lua/sample.lua
local json = require("json")
local utils = require("utils")

function greet(name)
    return "Hello, " .. name .. "!"
end

function add(a, b)
    return a + b
end

local function square(x)
    return x * x
end

local function clamp(value, min, max)
    if value < min then return min end
    if value > max then return max end
    return value
end

local function main()
    local msg = greet("World")
    local sum = add(1, 2)
    local sq = square(sum)
    print(msg)
end
